### PR TITLE
[ISSUE-42] feat: jaccard simiar

### PR DESCRIPTION
**Pull request by kaori-seasons**
### What changes were proposed in this pull request?
related to [issues-42](https://github.com/TuGraph-family/tugraph-analytics/issues/42)

### How was this PR tested?
- [ ] Tests have Added for the changes
- [ ] Production environment verified

### CODE REVIEW

File: geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                }
+            }
+        }
+    }
+
+    @Override
+    public void finish(RowVertex vertex) {
+        long srcId = (long) vertex.getValue().getField(0, LongType.INSTANCE);
+        long targetId =  (long) vertex.getValue().getField(1, LongType.INSTANCE);
+        double similar =  (double) vertex.getValue().getField(1, DoubleType.INSTANCE);
```
Comment (Leomrlin) [inline geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java]: Why is similar retrieved from the field at offset 1?

File: geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        List<RowEdge> bothEdges = new ArrayList<>(context.loadEdges(EdgeDirection.BOTH));
+        List<String> collection = bothEdges.stream().map(x -> String.valueOf(x.getTargetId())).collect(Collectors.toList());
+
+        if (context.getCurrentIterationId()  == iteration) {
+
+            bothEdges.stream().forEach(x -> {
+                //Update the value of the current iteration table
+                String initValueStr = collection.toString();
+                initValueStr = initValueStr.substring(1, initValueStr.length()-1);
```
Comment (Leomrlin) [inline geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java]: Why do we need to call substring to remove the first character?

File: geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            while (messages.hasNext()) {
+                ObjectRow singleRow = (ObjectRow) messages.next();
+                // set of neighbors
+                Long neighborsVertexId =  (Long) singleRow.getField(0, LongType.INSTANCE);
+                String vertexCollection = (String) singleRow.getField(1, StringType.INSTANCE);
+                if (StringUtils.isBlank(vertexCollection)) {
+                    return;
+                } else {
+                    String[] vertexIds =  vertexCollection.split(",");
+                    List<Long> neighborsVertexCollection =  Arrays.asList(vertexIds).stream().map(Long::valueOf).collect(Collectors.toList());
```
Comment (Leomrlin) [inline geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java:85]: The class of vertex id may not be Long, use Object instead.

**Review by Leomrlin — APPROVED** [review]
Thank you for implementing the similarity algorithm. I have provided some comments, could you please take a look at them?

File: geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java (hunk 3 above)
Comment (kaori-seasons) [reply geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java:85]: @Leomrlin Hello, I'm a bit confused. In the test dataset, the table definition vertexId is of type Long. Why do I need to use the Object type?

**Pull request closed without merging**